engines/agos/input.cpp

```diff
@@ -573,8 +574,8 @@ bool AGOSEngine::processSpecialKeys() {
 	if (shouldQuit())
 		_exitCutscene = true;
 
-	switch (_keyPressed.keycode) {
-	case Common::KEYCODE_UP:
+	switch (_action) {
+	case AGOSAction::kActionWalkForward:
```

Comment:
I recommend dropping the enum name
```suggestion
	case kActionWalkForward:
```

and do it in all instances